### DISCUSSION TITLE
https://theme-hope.vuejs.press/zh/config/plugins/search.html

### DISCUSSION THREAD

**Discussion by giscus** [vuepress-theme-hope/giscus-discussions#849] [Announcements]
# https://theme-hope.vuejs.press/zh/config/plugins/search.html

I介绍 主题内置支持以下三个插件: @vuepress/plugin-docsearch vuepress-plugin-search-pro @vuepress/plugin-search 你只能同时启用其中的 1 个，优先级为 docsearch > searchPro > search。 plugins.docsearch 传递给 @vuepre...

https://theme-hope.vuejs.press/zh/config/plugins/search.html

**Comment by zhuonixian** [comment]
plugins.searchPro 有完整的项目 config.ts 参考吗 

增加后直接报Error了..

**Reply by Mister-Hope** [reply]
原则上你直接安装插件后加一个searchPro: true就行

**Comment by AlanNandy** [comment]
请问本地运行docsearch正常，在GitHub上部署提示ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up to date with package.json，可以怎么解决呢

**Reply by AlanNandy** [reply]
发现了……不要用npm改用pnpm，因为部署是用pnpm安装依赖的

**Reply by AlanNandy** [reply]
如果爬虫正常但是前端不显示索引，则需要额外安装docsearch/js：
`pnpm add -D @docsearch/js@3`

**Comment by xiaomaohuifaguang** [comment]
docSearch 我部署在github 不是根目录 base是blog 现在搜索好使 但是点击跳转的url 会多一个base
例如 https://..../blog/blog/模板.html 正确的应该是
